### PR TITLE
RAS: Improve header spacing when My Account link is visible

### PR DESCRIPTION
**Pull request by laurelfulford**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR attempts to create some more space for the RAS My Account link, without changing very much about the spacing and appearance of the header when the link is not enabled. 

To hopefully avoid causing too many changes to live sites, I tried to target these changes to only be applied when the logo is centred or the header is short, and when the My Account link is enabled. 

See 203981098043663-as-1203870203438177

### How to test the changes in this Pull Request:

This is one of those pull requests that ideally will be tested with a bunch of different header settings, and different logo shapes (a vertical logo, and a horizontal logo). 

<details><summary><strong>Some tester logos if you'd like them!</strong></summary>

logo-1

logo-2

logo-3

logo-4
</details>

I've tried to break this down into different specific things to test, if you really want to get into the weeds -- I've gone through all of these as well, so it's really just to help spook out anything I may have missed! 😬 

This PR should be tested with https://github.com/Automattic/newspack-plugin/pull/2311.

#### Code

1. Is the approach for checking for RAS a good one, or is there another way we'd prefer this be handled?

#### Comparing non-RAS views

1. Turn off RAS.
2. Cycle through the following combinations and compare master to this PR - the vertical spacing may differ a bit, but the logo sizes should be the same:

- [ ] Default height with left aligned logo
- [ ] Default height with left aligned logo + sticky
- [ ] Default height with centre aligned logo
- [ ] Default height with centre aligned logo + sticky
- [ ] Short height with a left aligned logo
- [ ] Short height with a left aligned logo + sticky
- [ ] Short height with a centre aligned logo
- [ ] Short height with a centre aligned logo + sticky

<details><summary><strong>Screenshots - Default Height Header</strong></summary>

In these screenshots, you can see the slight spacing difference in the tertiary menu, but the logo should be the same size in the before and after.

I set the logos to a larger, but reasonable size (it is set to a smaller size for the short header).

Default header, centred logo - before
default-centre-master

Default header, centred logo - after
default-centre

Default header, centred logo, sticky - before
default-centre-sticky-master

Default header, centred logo, sticky - after
default-centre-sticky

Default header, left logo - before
default-left-master

Default header, left logo - after
default-left

Default header, left logo, sticky - before
default-left-sticky-master

Default header, left logo, sticky - after
default-left-sticky
</details>

<details><summary><strong>Screenshots - Short Height Header</strong></summary>

In these screenshots, you can see the slight spacing difference in the tertiary menu, but the logo should be the same size in the before and after.

I set the logos to a larger, but reasonable size (it is set to a smaller size for the short header).

Short header, centre logo - before
short-centre-master

Short header, centre logo - after
short-centre

Short header, centre logo, sticky - before
short-centre-sticky-master

Short header, centre logo, sticky - after
short-centre-sticky

Short header, left logo - before
short-left-master

Short header, left logo - after
short-left

Short header, left logo, sticky - before
short-left-sticky-master

Short header, left logo, sticky - after
short-left-sticky
</details>

#### Comparing RAS views

1. Turn on RAS.
2. Apply this PR.
3. Cycle through the following combinations and confirm that non-sticky headers have a new max-width of 350px for the logos.
4. Try scaling down each view from desktop to mobile, and confirm that your menus _usually_ have enough space, and that things look okay. This isn't going to work with every combination of longer menus, but your tertiary menu + My Account link should work pretty well before it switches to the mobile view:

- [ ] Default height with centre aligned logo
- [ ] Default height with centre aligned logo + sticky
- [ ] Short height with a left aligned logo
- [ ] Short height with a left aligned logo + sticky
- [ ] Short height with a centre aligned logo
- [ ] Short height with a centre aligned logo + sticky

<details>
<summary><strong>Screenshots - Default Height Header</strong></summary>

In testing these header settings, it's expected that the default height, centred logo should be smaller than the non-RAS example above and their "before" counterparts. 

Default height, centre logo -before
ras-default-centre-master

Default height, centre logo - after 
ras-default-centre

Default height, centre logo, sticky - before
ras-default-centre-sticky-master

Default height, centre logo, sticky - after
ras-default-centre-sticky

Default height, mobile - before
ras-default-centre-mobile-master

Default height, mobile - after
ras-default-left-mobile
</details>

<details>
<summary><strong>Screenshots - Short Height Header</strong></summary>

It's expected that the short header should have a smaller logo than the non-RAS screenshots above, for both the centred and left aligned options. 

Like with the default height header, the logos in the sticky headers should be roughly the same as their non-RAS counterparts.

Short header, centre logo - before:
ras-short-centre-master

Short header, centre logo - after:
ras-short-centre

Short header, centre logo, sticky - before:
ras-short-centre-sticky-master

Short header, centre logo, sticky - after:
ras-short-centre-sticky

Short header, left logo - before:
ras-short-left-master

Short header, left logo - after:
ras-short-left

Short header, left logo, sticky - before:
ras-short-left-sticky-master

Short header, left logo, sticky - after:
ras-short-left-sticky

Short header, mobile - before:
ras-short-centre-mobile-master

Short header, mobile - after:
ras-short-left-mobile

</details>

#### Awesome AMP testing that everyone loves

1. Turn on RAS
2. Apply this PR
3. Spot check a few of the header options below with AMP and without AMP -- you're specifically checking:
* With AMP enabled: do smaller/portrait-sized logos have excessive space around them?
* With AMP disabled: are logos of any size/shape distorted?

Randomly testing and resizing some of these settings with a tall + small logo, and a horizontal one, would be much appreciated! 

- [ ] Default height with centre aligned logo
- [ ] Default height with centre aligned logo + sticky
- [ ] Short height with a left aligned logo
- [ ] Short height with a left aligned logo + sticky
- [ ] Short height with a centre aligned logo
- [ ] Short height with a centre aligned logo + sticky

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — CHANGES_REQUESTED** [review]
Most of the combinations I tested worked well and provided enough space for the Sign In/My Account button.

This combination doesn't work so well, but it probably has more elements than a real site would (short header + long horizontal logo + show site title + show site tagline + longish menu + RAS). 

Screen Shot 2023-03-08 at 2 48 05 PM

I did find that adding `flex: 1;` to the `.site-branding` element fixed it in this case.

Screen Shot 2023-03-08 at 3 03 52 PM

Also left a few suggestions about the code checking whether RAS exists/is active.

**Comment by laurelfulford** [comment]
Thanks for reviewing this, @dkoo!

In revisiting this, I'm seeing that it's going to have more of an impact on RAS sites that have already launched than I expected 😬 I'm going to work on your updates, as well as trying to reign things in a bit so they don't affect logos sizes on desktop as much.

File: newspack-theme/inc/template-functions.php [hunk 1]
```diff
@@ -164,6 +164,14 @@ function newspack_body_classes( $classes ) {
 		$classes[] = 'has-highlight-menu';
 	}
 
+	// Adds class if the RAS account link is enabled.
+	if ( class_exists( '\Newspack\Reader_Activation' ) ) {
```
[the remark this answers is not in the document]
Comment (laurelfulford) [reply newspack-theme/inc/template-functions.php]: This should be addressed in [cacc4f7](https://github.com/Automattic/newspack-theme/pull/2051/commits/cacc4f72866fa0e83c3a1dc6d7a1c1eed2f0ec0f)!

File: newspack-theme/inc/template-functions.php [hunk 2]
```diff
@@ -164,6 +164,14 @@ function newspack_body_classes( $classes ) {
...
+		$ras_settings = \Newspack\Reader_Activation::get_settings();
+		if ( $ras_settings['enabled'] && $ras_settings['enabled_account_link'] ) {
```
[the remark this answers is not in the document]
Comment (laurelfulford) [reply newspack-theme/inc/template-functions.php]: This should be addressed in [cacc4f7](https://github.com/Automattic/newspack-theme/pull/2051/commits/cacc4f72866fa0e83c3a1dc6d7a1c1eed2f0ec0f)!

File: newspack-theme/inc/logo-resizer.php [hunk 3]
```diff
@@ -67,6 +67,16 @@ function newspack_customize_logo_resize( $html ) {
 
 		$logo_max_width = ( $logo['width'] > 600 ) ? 600 : $logo['width'];
 
+		// Update the max-width to match the RAS max-width in the CSS - this improves the Customizer preview accuracy.
+		if ( class_exists( '\Newspack\Reader_Activation' ) ) {
+			$header_simplified     = get_theme_mod( 'header_simplified', false );
+			$header_center_logo    = get_theme_mod( 'header_center_logo', false );
+			$ras_settings = \Newspack\Reader_Activation::get_settings();
+			if ( $ras_settings['enabled'] && $ras_settings['enabled_account_link'] && ( $header_simplified || $header_center_logo ) ) {
```
[the remark this answers is not in the document]
Comment (laurelfulford) [reply newspack-theme/inc/logo-resizer.php]: I ended up removing this code from the logo-resizer.php file, so this should be resolved, but for different reasons.

File: newspack-theme/inc/logo-resizer.php [hunk 4]
```diff
@@ -67,6 +67,16 @@ function newspack_customize_logo_resize( $html ) {
 
 		$logo_max_width = ( $logo['width'] > 600 ) ? 600 : $logo['width'];
 
+		// Update the max-width to match the RAS max-width in the CSS - this improves the Customizer preview accuracy.
+		if ( class_exists( '\Newspack\Reader_Activation' ) ) {
```
[the remark this answers is not in the document]
Comment (laurelfulford) [reply newspack-theme/inc/logo-resizer.php]: Thanks for these, @dkoo! I ended up removing this code from the logo-resizer.php file, so this should be resolved, but for different reasons.

**Review by dkoo — APPROVED** [review]
New RAS checks and methods look 👍 

The combo I called out before (short header + long horizontal logo + show title + show tagline + RAS My Account button) looks better than before, but differs slightly from the [original style](https://github.com/Automattic/newspack-theme/pull/2051#pullrequestreview-1331180418) in which the title and tagline sit side-by-side. It doesn't look worse now, just different, so I'm approving.

Screen Shot 2023-03-28 at 9 44 24 AM

**Comment by adekbadek** [comment]
@laurelfulford – what's the status on this one?

**Comment by laurelfulford** [comment]
@adekbadek thanks for following up on this! I did some more of my own testing, and wasn't entirely confident this wasn't going to cause unexpected outcomes with live site headers 😬 It's been long enough now, I wonder if it'd be best just to tighten up the spacing/reduce the button size a bit, and leave the logo alone (which is where I ran into the most noticeable differences).

**Comment by laurelfulford** [comment]
I'm going to close this PR and start from scratch -- the spacing could be improved, but messing with the logo sizing at this point is too risky.

**Pull request closed without merging**